### ISSUE TITLE
support non-desktop components

### ISSUE THREAD

**Issue by bilelmoussaoui** [flathub/backend#9]
Currently, the api endpoints are desktop-app specific. We should expose at least cli, extensions & sdk's. This also would be a step forward to support displaying extensions of an application.

**Comment by tchx84** [comment]
> Currently, the api endpoints are desktop-app specific. We should expose at least cli, extensions & sdk's. This also would be a step forward to support displaying extensions of an application.

How do you think the API should look like?

Here's the thing:
1. CLI tools are just regular apps without appstream.xml nor .desktop.

Should we expose CLI tools under the same apps endpoint, and simply extend the details struct with a "cli: true" field? Or should we provide a separate endpoint? My first thought is to go for the first option as it would simplify a lot of things, e.g. we ca re-use the other existing endpoints like search.

2. Extensions and SDKs are both runtimes. Any clues on how we want to expose these? Are these complementary to apps? e.g. they will be displayed along with apps, or should the be presented / searchable independently ?

**Comment by barthalion** [comment]
Some more questions: appstream-glib is picky about console applications shipping icons. We need that changed too.

**Comment by razzeee** [comment]
Extensions are now exposed via the summary endpoint or is there more, that I'm not aware of?
So we could show them on the detail page, would there be an action we can take?

Also is there more data then this? A nicer name for e.g.?
```
   "extensions": {
      "tv.kodi.Kodi.Debug": {
        "directory": "lib/debug",
        "autodelete": "true",
        "no-autodownload": "true"
      }
    },
```

**Comment by razzeee** [comment]
Some small changes make this work on our end

![image](https://user-images.githubusercontent.com/5943908/144325844-acfc3e29-6ba4-4478-9a1d-170df96a3043.png)

![image](https://user-images.githubusercontent.com/5943908/144325916-59aefbe7-dab0-4773-a8b4-3d5f9bee73c9.png)

This is only with `console-applications` in addition. 18 apps more currently, so I checked all, they work fine, besides the logo. Which we could just not show on the detail page and use the flathub logo/a placeholder on the listings.

**Comment by bilelmoussaoui** [comment]
>  use the flathub logo/a placeholder on the listings.

I don't think that's a good approach. People will end up confusing what the app is about and think it's an "offical Flathub app" or whatever.

I think CLI apps needs a different design/categories in general; different from desktop apps.

**Comment by razzeee** [comment]
> I don't think that's a good approach. People will end up confusing what the app is about and think it's an "offical Flathub app" or whatever.

makes sense for the flathub logo, but having a generic package logo shouldn't cause that problem. we could also overlay the app name onto the image, to help with people thinking they might be the same.

> I think CLI apps needs a different design/categories in general; different from desktop apps.

While that would solve the problem, I think it would create new ones, which I think would be worse. Search for example would need to have two listings with both of those different designs.


That being said, obviously, having the logos in the first place is the best approach

**Comment by razzeee** [comment]
This also needs something like a tag on the detail page, to let users easily understand, that it's a cli app

**Comment by bilelmoussaoui** [comment]
> This also needs something like a tag on the detail page, to let users easily understand, that it's a cli app

The issue here is that it's not only about CLI apps (unless that's explicitly your goal) but we also have SDKs/Runtimes/Extensions that needs to be showed somehow :)

**Comment by razzeee** [comment]
I think it makes sense to break this up. We probably want them to show up differently, especially extensions. And it might be worth it getting the parts in, that we can already solve.

**Comment by abique** [comment]
> I think it makes sense to break this up. We probably want them to show up differently, especially extensions. And it might be worth it getting the parts in, that we can already solve.

Yes you're right.
In regards to Digital Audio Workstation, I could even imagine a dedicated category for VST/LV2 plugins.
As of today there aren't so many VST plugins on flathub, but if it takes off, it can really grow massively.
